streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -1576,139 +1562,49 @@ public void shouldProcessFromSourcesThatMatchMultiplePattern() {
     }
 
     @Test
-    @SuppressWarnings("deprecation")
-    public void shouldNotAllowBadTransformerSupplierOnFlatTransform() {
-        final org.apache.kafka.streams.kstream.Transformer<String, String, Iterable<KeyValue<String, String>>> transformer = flatTransformerSupplier.get();
+    public void shouldNotAllowBadProcessSupplierOnProcess() {
```

Comment:
Is there no existing test for `ProcessSupplier` to test test case? Just wondering if this rewrite makes sense (and closes a test gap), or if this rewrite would duplicate an existing test (and thus, we could just remove this test instead of rewriting it)?